models/Element/Service.php

```diff
@@ -1148,9 +1148,8 @@ public static function getValidKey($key, $type)
             $key = ltrim($key, '. ');
         }
 
-        $key = mb_substr($key, 0, 255);
-
-        return $key;
+        // key should not end with space after cut
+        return trim(mb_substr($key, 0, 255));
```

Comment:
Should we use `rtrim` if we only care about removing whitespace at the end?

```suggestion
        return rtrim(mb_substr($key, 0, 255));
```

Otherwise, I think we should adapt the comment.

Reply:
Good point, thanks,
my guess on adapt a comment, better be sure that there are no spaces on start and end, 
no matter what is going on before

Reply:
Yes, bc of mb_substr() function we rarely get this exception on a big strings.
ex: I'm importing few millions of products and using title for a key and got ~100 random exceptions on that,
currently I'm using hack `trim(Service::getValidKey($id.$title, 'object'))`